client/asset/xmr/toolsdl/download.go

```diff
@@ -0,0 +1,749 @@
+//////////////////////////////////////////////////////////////////////////go:build xmrlive
+
+package toolsdl
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"bufio"
+	"compress/bzip2"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"decred.org/dcrdex/dex"
+)
+
+// The toolsdl package is a method to download monero-wallet-cli and monero-wallet-rpc
+// binaries from getmonero.org.
+//
+// First the canonical signed hashes.txt file is downloaded and  checked for version and
+// if needed compared to current. Then the list of zipfiles with hashes is examined for the
+// one containing the hashed zipfile for the architecture the end user is on.
+//
+// That file (tar/bz2 or zip) is checked against it's sha256 hash. If correct the file is
+// decompressed and the 2 specific binaries above are extracted to the user directory
+// from where they can be subsequently run.
+
+// This is the relevant part of hashes.txt:
+//
+// ## CLI
+// 7c2ad18ca3a1ad5bc603630ca935a753537a38a803e98d645edd6a3b94a5f036  monero-android-armv7-v0.18.4.4.tar.bz2
+// eb81b71f029884ab5fec76597be583982c95fd7dc3fc5f5083a422669cee311e  monero-android-armv8-v0.18.4.4.tar.bz2
+// bc539178df23d1ae8b69569d9c328b5438ae585c0aacbebe12d8e7d387a745b0  monero-freebsd-x64-v0.18.4.4.tar.bz2
+// 2040dc22748ef39ed8a755324d2515261b65315c67b91f449fa1617c5978910b  monero-linux-armv7-v0.18.4.4.tar.bz2
+// b9daede195a24bdd05bba68cb5cb21e42c2e18b82d4d134850408078a44231c5  monero-linux-armv8-v0.18.4.4.tar.bz2
+// c939ea6e8002798f24a56ac03cbfc4ff586f70d7d9c3321b7794b3bcd1fa4c45  monero-linux-riscv64-v0.18.4.4.tar.bz2
+// 7fe45ee9aade429ccdcfcad93b905ba45da5d3b46d2dc8c6d5afc48bd9e7f108  monero-linux-x64-v0.18.4.4.tar.bz2
+// 8c174b756e104534f3d3a69fe68af66d6dc4d66afa97dfe31735f8d069d20570  monero-linux-x86-v0.18.4.4.tar.bz2
+// 645e9bbae0275f555b2d72a9aa30d5f382df787ca9528d531521750ce2da9768  monero-mac-armv8-v0.18.4.4.tar.bz2
+// af3d98f09da94632db3e2f53c62cc612e70bf94aa5942d2a5200b4393cd9c842  monero-mac-x64-v0.18.4.4.tar.bz2
+// 7eb3b87a105b3711361dd2b3e492ad14219d21ed8fd3dd726573a6cbd96e83a6  monero-win-x64-v0.18.4.4.zip
+// a148a2bd2b14183fb36e2cf917fce6f33fb687564db2ed53193b8432097ab398  monero-win-x86-v0.18.4.4.zip
+// 84570eee26238d8f686605b5e31d59569488a3406f32e7045852de91f35508a2  monero-source-v0.18.4.4.tar.bz2
+// #
+
+const (
+	HashesLink           = "https://www.getmonero.org/downloads/hashes.txt"
+	MoneroHashesFilename = "hashes.txt"
+	DexMoneroToolsPath   = ".dexc/share/monero-tools"
```

Comment:
This doesn't work because the user can change the path. You have to get the path from core at some point for the app dir.